patches/minecraft/net/minecraft/client/gui/screens/controls/KeyBindsList.java.patch

```diff
@@ -22,11 +44,19 @@
              KeyBindsList.this.f_93386_.f_91066_.m_92159_(p_193916_, p_193916_.m_90861_());
              KeyBindsList.this.m_269130_();
           }).m_252987_(0, 0, 50, 20).m_252778_((p_253313_) -> {
-@@ -144,7 +_,7 @@
+@@ -143,8 +_,14 @@
+ 
        public void m_6311_(GuiGraphics p_281805_, int p_281298_, int p_282357_, int p_281373_, int p_283433_, int p_281932_, int p_282224_, int p_282053_, boolean p_282605_, float p_281432_) {
           int k = p_281373_ + 90 - KeyBindsList.this.f_193859_;
-          p_281805_.m_280614_(KeyBindsList.this.f_93386_.f_91062_, this.f_193911_, k, p_282357_ + p_281932_ / 2 - 9 / 2, 16777215, false);
+-         p_281805_.m_280614_(KeyBindsList.this.f_93386_.f_91062_, this.f_193911_, k, p_282357_ + p_281932_ / 2 - 9 / 2, 16777215, false);
 -         this.f_193913_.m_252865_(p_281373_ + 190);
++         // NeoForge: Trim strings that are too long, and show a tooltip if the mouse is over the trimmed string
++         List<net.minecraft.network.chat.FormattedText> lines = KeyBindsList.this.f_93386_.f_91062_.m_92865_().m_92414_(this.f_193911_, NAME_SPLIT_LENGTH, net.minecraft.network.chat.Style.f_131099_);
++         MutableComponent nameComponent = Component.m_237113_(lines.get(0).getString() + (lines.size() > 1 ? "..." : ""));
++         if(lines.size() > 1 && this.m_5953_(p_282224_ + 95, p_282053_) && p_282224_ < p_281373_ - 90 + KeyBindsList.this.f_193859_) {
++            KeyBindsList.this.f_93386_.f_91080_.m_257959_(Tooltip.m_257868_(KeyBindsList.this.f_93386_, this.f_193911_));
```

Comment:
Instead of `KeyBindsList.this.minecraft.screen.setTooltipForNextRenderPass` I believe we can just do `KeyBindsList.this.keyBindsScreen.setTooltipForNextRenderPass`.

Part of me also wonders if we want to use `Tooltip#splitTooltip` or given we already split it do we want to just do the last step that `Tooltip#splitTooltip` would do of `Language.getInstance().getVisualOrder(lines)`... Yes this would make it use a width of 185 rather than 170 for the split so it would be slightly functionally different than the current patch implements it, but still might make some sense

Reply:
Switching the line to
`List<net.minecraft.util.FormattedCharSequence> lines = KeyBindsList.this.minecraft.font.split(this.name, NAME_SPLIT_LENGTH);

And then using KeyBindsList.this.keyBindsScreen.setTooltipForNextRenderPass(lines); would also work

Reply:
![image](https://github.com/neoforged/NeoForge/assets/3209834/24a16042-0014-41ad-9937-e3ab7bc0cbe8)
Thats how it looked with those changes